src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -126,15 +126,47 @@ HashSet<ITypeSymbol> typeStack
 			// sufficiently immutable, and we also need to ensure the
 			// immutability of the type parameters is also appropriate.
 			ImmutabilityScope scope = type.GetImmutabilityScope();
-			if( !type.IsGenericType()
-				&& !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute )
+			if( !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute )
 				&& scope != ImmutabilityScope.None
 			) {
+				if( type.IsGenericType() ) {

```

Comment:
We get in here if it's marked immutable, so now we need to perform the additional check on the type parameters that we weren't before.  Specifically, check if the parameter is supposed to be immutable, then check if the type evaluates to immutable.